android/app/src/main/kotlin/coop/polypoly/polypod/updatenotification/UpdateNotification.kt

```diff
@@ -5,23 +5,65 @@ import coop.polypoly.polypod.Preferences
 import coop.polypoly.polypod.R
 
 class UpdateNotification(private val context: Context) {
-    val id = context.resources.getInteger(R.integer.update_notification_id)
+    companion object {
+        data class MockData(var id: Int? = null)
+        val mockData = MockData()
+    }
+
+    private enum class State {
+        NOT_SEEN,
+        PUSH_SEEN,
+        ALL_SEEN;
+
+        companion object {
+            fun parse(s: String?) =
+                s.let { name -> values().firstOrNull { it.name == name } }
+        }
+    }
+
+    val id = mockData.id
+        ?: context.resources.getInteger(R.integer.update_notification_id)
+
     val title = context.getString(R.string.update_notification_title)
+
     val text = context.getString(R.string.update_notification_text)
+
     val pushDelay =
         context.resources.getInteger(R.integer.update_notification_push_delay)
 
-    val inAppNotificationSeen: Boolean
-        get() = id == 0 || Preferences.getSeenInAppNotificationId(context) == id
+    val showPush: Boolean
+        get() = state == State.NOT_SEEN
+
+    val showInApp: Boolean
+        get() = state != State.ALL_SEEN
 
-    val pushNotificationSeen: Boolean
-        get() = id == 0 || Preferences.getSeenPushNotificationId(context) == id
+    private var state: State = loadLastState()
+        set(value) {
+            field = value
+            Preferences.setLastNotification(context, id, state.name)
+        }
+
+    private fun loadLastState(): State {
+        val (lastId, lastState) = Preferences.getLastNotification(context)
+        return when (id) {
+            0 -> State.ALL_SEEN
+            lastId -> State.parse(lastState) ?: State.ALL_SEEN
```

Comment:
So there are two paths to `ALL_SEEN`: id == 0 or a non-integer value in lastState. So if you set the id to "FOOBAR" or "0XDEADBEEF" if would also lead to the same result, right? Wouldn't that be a bit better than using a specific state ID for the same purpose?

Reply:
It wouldn't work because it's an integer resource, but if I turned it into a string that I parsed a number from, it would!

Reply:
So State.parse would raise an exception if it's not a string containing an integer? So what you're doing here is essentially if it parses to anything that's not 0, set to that state, if it parses to 0, State.ALL_SEEN, correct?

Reply:
Ah no, the states are actually stored as strings: `"ALL_SEEN"`, `"NOT_SEEN"` or `"PUSH_SEEN"` respectively. `0` is the _id_, not the _state_.

If `State.parse` finds anything _other_ than those values, it returns `null,` in which case this line here would return `ALL_SEEN`.